kafka/conn.py

```diff
@@ -674,7 +674,22 @@ def _try_authenticate_plain(self, future):
         log.info('%s: Authenticated as %s via PLAIN', self, self.config['sasl_plain_username'])
         return future.success(True)
 
-    def _try_authenticate_aws_msk_iam(self, future, session):
+    def _try_authenticate_aws_msk_iam(self, future, role_arn=None):
+        if role_arn:
+            client = BotoSession().client('sts')
+            assume_role = client.assume_role(
+                RoleArn=role_arn,
+                RoleSessionName="kafka-python"
+            )
+            credentials = assume_role['Credentials']
+            session = BotoSession(
+                aws_access_key_id=credentials['AccessKeyId'],
+                aws_secret_access_key=credentials['SecretAccessKey'],
+                aws_session_token=credentials['SessionToken'],
+            )
```

Comment:
This should go into a separate private method or nested method, but that is a personal preference, not a requirement :)

Reply:
Finally, I moved the entire Session creation logic to a private method.